src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -63,7 +63,9 @@ def delete_func(resource_group_name, resource_name, item_name):
         keep_items = \
             [x for x in item.__getattribute__(prop) if x.name.lower() != item_name.lower()]
         _set_param(item, prop, keep_items)
-        return client.create_or_update(resource_group_name, resource_name, item)
+        result = client.create_or_update(resource_group_name, resource_name, item).result()
+        if next((x for x in getattr(result, prop) if x.name == item_name), None):
```

Comment:
Please consider whether the `item_name` could have casing difference. If yes, do case insensitive comparison